virt_lightning/virt_lightning.py

```diff
@@ -214,53 +235,73 @@ def kvm_binary(self):
 
-    def init_network(self, bridge_name):
+    def init_network(self, network_name, network_cidr):
         try:
-            bridge_netiface = netifaces.ifaddresses(bridge_name)
-        except ValueError:
-            print("Bridge not found:", bridge_name)
-            exit(1)
-        ipconfig = bridge_netiface[netifaces.AF_INET]
-        self.gateway = ipaddress.IPv4Interface("{addr}/{netmask}".format(**ipconfig[0]))
+            self.network_obj = self.conn.networkLookupByName(network_name)
+        except libvirt.libvirtError as e:
+            if e.get_error_code() == libvirt.VIR_ERR_NO_NETWORK:
+                pass
+
+        if not self.network_obj:
+            self.network_obj = self.create_network(network_name, network_cidr)
+
+        if not self.network_obj.isActive():
+            self.network_obj.create(0)
+
+        xml = self.network_obj.XMLDesc(0)
+        root = ET.fromstring(xml)
+        ip = root.find("./ip")
+
+        self.gateway = ipaddress.IPv4Interface(
+            "{address}/{netmask}".format(**ip.attrib)
+        )
         self.dns = self.gateway
         self.network = self.gateway.network
 
+    def create_network(self, network_name, network_cidr):
+        network = ipaddress.ip_network(network_cidr)
+        root = ET.fromstring(NETWORK_XML)
+        root.find("./name").text = network_name
+        root.find("./bridge").attrib["name"] = network_name
+        root.find("./ip").attrib = {
+            "address": network[1].exploded,
+            "netmask": network.netmask.exploded,
+        }
+        xml = ET.tostring(root).decode()
+        return self.conn.networkCreateXML(xml)
+
     def init_storage_pool(self, storage_pool):
         try:
             self.storage_pool_obj = self.conn.storagePoolLookupByName(storage_pool)
-            if not self.get_storage_dir().is_dir():
-                raise Exception("Missing storage directory:", self.get_storage_dir())
-            if not self.storage_pool_obj.isActive():
-                self.storage_pool_obj.create(0)
-            return
         except libvirt.libvirtError as e:
             if e.get_error_code() == libvirt.VIR_ERR_NO_STORAGE_POOL:
                 pass
 
-        if self.conn.getURI().startswith("qemu:///system"):
+        if not self.storage_pool_obj:
             storage_dir = pathlib.PosixPath("/var/lib/virt-lightning/pool")
-        elif self.conn.getURI().startswith("qemu:///session"):
-            storage_dir = pathlib.PosixPath("~/.local/share/virt-lightning/pool")
-            storage_dir = storage_dir.expanduser()
-        else:
-            raise Exception("Unsupported libvirt URI")
+            self.storage_pool_obj = self.create_storage_pool(
+                name=storage_pool, directory=storage_dir
+            )
 
-        full_dir = storage_dir / "upstream"
         try:
-            full_dir.mkdir(parents=True, exist_ok=True)
+            full_dir = self.get_storage_dir() / "upstream"
```

Comment:
Looks like bug. `str` can not divide another `str`.

Reply:
/ is a specialized operator of PosixPath. It is used to build paths.